doc/doxygen/src/kconfig/kconfig.md

```diff
@@ -0,0 +1,366 @@
+configurations (based on the used modules) for the particular platform will be
+presented. By default the configuration of a module via Kconfig is not enabled.
+In order to activate the configuration via Kconfig the correspondent option
+should be selected. That will enable the configuration of all inner options, if
+available.
+
+Once the desired configuration is achieved save the configuration to the
+default proposed path and exit. The saved configuration will be applied when
+the code is compiled (`make all`).
+
+If the current configuration will be used in the future it can be saved in the
+application's folder as `user.config`, and it will be persistent after cleaning
+the application directory via `make clean`.
+
+## Configure using '.config' files                     {#configure-using-files}
+The second way to configure the application is by directly writing '.config'
+files. Two files will be sources of configuration during the generation of the
+final header file: `app.config` and `user.config`, which should be placed
+inside the application's folder. As the order matters, configurations from
+`app.config` can be overridden by `user.config`.
+
+Let's say that the `SOCK_UTIL_SCHEME_MAXLEN` symbol in `sock_util` module needs
+to be configured. The `user.config` file could look like:
+
+```Makefile
+# activate configuration of sock_util using Kconfig
+CONFIG_KCONFIG_MODULE_SOCK_UTIL=y
+# change scheme part length
+CONFIG_SOCK_UTIL_SCHEME_MAXLEN=24
+```
+
+In this case, there is no need for using menuconfig. It's enough just to call
+`make all` in the application folder, as this configuration will be read and
+applied. Note that if any dependency issue occurs, warnings will be generated
+(e.g. not enabling the configuration of a module via Kconfig).
+
+## Application configuration with Kconfig                 {#app-config-kconfig}
+To expose application-specific configuration options a `Kconfig` file can
+be placed in the application's folder. For an example of this you can check
+the `tests/kconfig` application.
+
+## A note on the usage of CFLAGS
+When a certain module is being configured via Kconfig the configuration macro
+will not longer be overridable by means of CFLAGS (e.g. set on the
+compilation command or on a Makefile). Consider this if you are getting a
+'redefined warning'.
+
+## A note on the usage of the 'clean' command
+When using Kconfig as the configurator for RIOT, configuration symbols may be
+used in Makefiles through the build system. For this to work properly make
+sure that when cleaning an application you call `make clean && make all`,
+instead of both targets on a single command.
+
+---
+# Integration into the build system    {#kconfig-integration-into-build-system}
+
+The integration of Kconfig into the build system is mainly done in
+`makefiles/kconfig.mk`.
+
+## Steps during the build process                {#kconfig-steps-build-process}
+
+![Ouput of every step](kconfig_integration.svg)
```

Comment:
The caption seems duplicate to the section heading. Either you be a bit more precise (I'd prefer that if there is more to say) or you get rid of it.